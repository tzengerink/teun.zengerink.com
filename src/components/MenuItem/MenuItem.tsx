import classNames from 'classnames'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { Project } from '../../lib/types'

interface MenuItemProps {
  project: Project
  onClick: () => void
}

const MenuItem: React.FC<MenuItemProps> = ({ project, onClick }) => {
  const router = useRouter()
  const isActive = router.asPath.includes(project.slug)

  return (
    <li key={project.slug} className={classNames('text-xs', { underline: isActive })}>
      <Link
        href={`/work/${project.slug}/${project.photos[0].key}`}
        className={classNames('hover:underline')}
        onClick={onClick}
      >
        {project.title}
      </Link>
    </li>
  )
}

export default MenuItem
