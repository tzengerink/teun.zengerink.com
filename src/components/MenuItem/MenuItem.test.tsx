import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import mockLink from '../../__mocks__/nextLink'
import mockProjects from '../../__mocks__/projects'
import MenuItem from './MenuItem'

const mockRouter = jest.fn(() => ({ asPath: `/` }))
jest.mock('next/router', () => ({ useRouter: () => mockRouter() }))

jest.mock('next/link', () => mockLink)

describe('MenuItem', () => {
  const project = mockProjects[0]
  const onClick = jest.fn()
  const renderComponent = () => render(<MenuItem project={project} onClick={onClick} />)

  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('renders correctly if project is active', () => {
    mockRouter.mockReturnValueOnce({ asPath: `/work/${project.slug}` })
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('properly dispatches on click', () => {
    const { getByText } = renderComponent()
    fireEvent.click(getByText(project.title))
    expect(onClick).toHaveBeenCalled()
  })
})
