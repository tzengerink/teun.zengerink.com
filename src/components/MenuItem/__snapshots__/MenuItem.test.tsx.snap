// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem renders correctly 1`] = `
<div>
  <li
    class="text-xs"
  >
    <a
      class="hover:underline"
      href="/work/my-first-project/01"
    >
      My First Project
    </a>
  </li>
</div>
`;

exports[`MenuItem renders correctly if project is active 1`] = `
<div>
  <li
    class="text-xs underline"
  >
    <a
      class="hover:underline"
      href="/work/my-first-project/01"
    >
      My First Project
    </a>
  </li>
</div>
`;
